src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestEntryPoint.ll

```diff
@@ -1,17 +1,36 @@
-define double @Microsoft__Quantum__Testing__QIR__TestEntryPoint(i64 %a__count, double* %a, i1 %b) #0 {
+define double @Microsoft__Quantum__Testing__QIR__TestEntryPoint({ i64, i8* }* %a, i8 %b) #0 {
 entry:
-  %0 = call %Array* @__quantum__rt__array_create_1d(i32 8, i64 %a__count)
-  %1 = icmp sgt i64 %a__count, 0
-  br i1 %1, label %copy, label %next
+  %0 = getelementptr { i64, i8* }, { i64, i8* }* %a, i64 0, i32 0
+  %1 = getelementptr { i64, i8* }, { i64, i8* }* %a, i64 0, i32 1
+  %2 = load i64, i64* %0
+  %3 = load i8*, i8** %1
+  %4 = call %Array* @__quantum__rt__array_create_1d(i32 8, i64 %2)
```

Comment:
It seems like all arrays received by entry-points are the same size (64-bits). For use cases in which the data-type of the array is not 64 bits (e.g. Pauli), should the inter-op array encode Pauli values in 64-bits?

Reply:
Coming back to this, I can no longer reproduce; passing a Pauli[] as entry point argument instead of a Double[] changes this line to   
```
%4 = call %Array* @__quantum__rt__array_create_1d(i32 1, i64 %2)
```
It seems that one of the other changes I did since you reviewed fixed the issue.